### PR TITLE
fix: shouldn't set input-bg inside a error form item

### LINKED ISSUE #25376: Antd4.x FormItem validate style influence disabled style
- [ ] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[Edit on CodeSandbox](https://codesandbox.io/s/antd-reproduction-template-0uiy5?file=/index.js)

### Steps to reproduce
Set the input in the form to disabled, and then conduct form verification. The form verification style will cover the disabled effect, which does not meet the expectation

将表单中的 input 置为 disabled，再进行表单校验，表单校验样式会把 disabled 的效果给覆盖，不符合预期

### What is expected?
The wrong style of the form item does not affect the native disabled style

表单项错误样式不影响原生disabled的样式

### What is actually happening?
The wrong style of form item affects the original disabled style, causing users' trouble

表单项错误样式影响原生disabled的样式，造成用户困扰

| Environment | Info |
|---|---|
| antd | 4.3.3 |
| React | 16 |
| System | mac |
| Browser | chrome 83 |

### PR DESCRIPTION
**Pull request by 07akioni**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link
fix https://github.com/ant-design/ant-design/issues/25376 

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |   Fix Input disabled `backgroundColor` is override when inside a `error` or `warning` FormItem.   |
| 🇨🇳 Chinese |   修正 Input 在禁用状态下背景颜色被 `error`  或 `warning` 的 FormItem 覆盖的问题。   |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/style/themes/default.less [hunk 1]
```diff
@@ -357,7 +357,6 @@
 // ---
 @label-required-color: @highlight-color;
 @label-color: @heading-color;
-@form-warning-input-bg: @input-bg;
```
Comment (zombieJ) [inline components/style/themes/default.less:360]: 有背景色变量看起来应该是非 disabled 下设置背景色

File: components/form/style/mixin.less [hunk 2]
```diff
@@ -10,21 +10,25 @@
...
     }
 
     &:focus,
     &-focused {
       .active(@border-color);
     }
+  }
 
-    &:not([disabled]):hover {
-      border-color: @border-color;
```
Comment (zombieJ) [inline components/form/style/mixin.less:23]: 这个去掉是因为没有影响吗？

Comment (07akioni) [reply components/form/style/mixin.less:23]: 是的，没用，form 造成的的优先级已经可以比 hover 这个要高了

Comment (07akioni) [reply components/form/style/mixin.less:23]: 例子里是不会出问题的，hover 之后

File: components/form/demo/disabled-input-debug.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
+---
+order: 30
+title:
+  zh-CN: Disabled Input Debug
+  en-US: Disabled Input Debug
```
Comment (afc163) [inline components/form/demo/disabled-input-debug.md:5]: 这个例子应该加个 `debug: true`。

**Pull request merged**